kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/AdminResource.scala

```diff
@@ -286,24 +288,38 @@ private[v1] class AdminResource extends ApiRequestContext with Logging {
     }
     val engine = normalizeEngineInfo(userName, engineType, shareLevel, subdomain, "default")
     val engineSpace = calculateEngineSpace(engine)
+    val responseMsgBuilder = new StringBuilder()
 
     withDiscoveryClient(fe.getConf) { discoveryClient =>
-      val engineNodes = discoveryClient.getChildren(engineSpace)
-      engineNodes.foreach { node =>
-        val nodePath = s"$engineSpace/$node"
+      val engineNodes = discoveryClient.getServiceNodesInfo(engineSpace, silent = true)
+      engineNodes.foreach { engineNode =>
+        val nodePath = s"$engineSpace/${engineNode.nodeName}"
+        val engineRefId = engineNode.engineRefId.orNull
         info(s"Deleting engine node:$nodePath")
```

Comment:
Sorry for the late comment. 
the `engineSpace` is not cleaned right? Why don't we delete it?

Reply:
except for CONNECTION level engine, we cannot guarantee that there are no other engines access the same `engineSpace`

Reply:
Do you mean there can be multiple engines under same subdomain?

Reply:
I don't think so. Even in GROUP level, we can only create one engine per subdomain, right. And subdomain has been included in `engineSpace` right?

Reply:
there are some corner cases. for example, if the engine has no response due to full GC or overload, kyuubi will create a new one on the same engine space. I also see sometimes the dist lock does not work properly(which should not, may be bugs), that also causes multi engines live in the same subdomain. finally, in theory, there are race conditions for kyuubi server deleting and engine registering under the same subdomain.

Reply:
Thanks for the explanation!